portal/assign.py

```diff
@@ -0,0 +1,171 @@
+@login_required
+@teacher_required
+
+def assign_create(sessions_id, course_id):
+    """Allows teachers to create new assignments for a
+    specific session"""
+
+    session = session_editor.get_session(sessions_id)
+    course = course_editor.get_course(course_id)
+
+    if g.user['id'] != course['teacher_id']:
+        return redirect(url_for('index'))
+
+    if course['course_num'] != session['course_id']:
+        return redirect(url_for('index'))
+
+    if request.method == 'POST':
+
+        name = request.form['name']
+        points = request.form['points']
+        description = request.form['description']
+        due_date = request.form['due_date']
+        error = None
+        with db.get_db() as con:
+            with con.cursor() as cur:
+
+                if not name:
+                    error = 'Name is required.'
+                try:
+                    int(points)
+                except ValueError:
+                    error = 'Points are numbers only, check your values.'
+                try:
+                    datetime.datetime.strptime(due_date, '%Y-%m-%dT%H:%M')
+                except ValueError:
+                    error = 'Due Date only allows time data, check your values.'
+
+                if error is None:
+                    now = datetime.datetime.utcnow()
+                    cur.execute("""INSERT INTO assignments (sessions_id, assign_name, description, points, due_time)
+                        VALUES (%s, %s, %s, %s, %s)
+                    """,
+                    (sessions_id, name, description, points, due_date, )
+                    )
+                    con.commit()
+
+                    return redirect(url_for("assign.assign_manage", sessions_id=session['id'], course_id=session['course_id'] ))
+
+                flash(error)
+
+    return render_template('layouts/assigns/assign_create.html', session=session)
+
+@bp.route('/course/<int:course_id>/session/<int:sessions_id>/assignments/', methods=('GET', 'POST'))
+@login_required
+@teacher_required
+
+def assign_manage(course_id, sessions_id):
+    """Allows teachers to see current assignments for a
+    specific session"""
+
+    session = session_editor.get_session(sessions_id)
+    course = course_editor.get_course(course_id)
+
+    if g.user['id'] != course['teacher_id']:
+        return redirect(url_for('index'))
+
+    if course['course_num'] != session['course_id']:
+        return redirect(url_for('index'))
+
+    cur=db.get_db().cursor()
+    cur.execute(
+        'SELECT * FROM assignments WHERE sessions_id = %s',
+        (sessions_id, )
+    )
+
+    assignments = cur.fetchall()
+
+    cur.close()
+
+    return render_template("layouts/assigns/assign_manage.html", assignments=assignments, session=session)
+
+@bp.route('/course/<int:course_id>/session/<int:sessions_id>/Edit/assignment/<int:assign_id>/', methods=('GET', 'POST'))
```

Comment:
here as well, if you put `/edit` at the end it will be easier to read and keep the URL more consistent.
So it would look like:
`/<int:sessions_id>/assignment`
`/<int:sessions_id>/assignment/create`
`/<int:sessions_id>/assignment/1/edit`
just keeps things organized nicely